src/modules/Sticky/Sticky.js

```diff
@@ -0,0 +1,251 @@
+import _ from 'lodash'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+import {
+  customPropTypes,
+  META,
+  getElementType,
+  getUnhandledProps,
+} from '../../lib'
+import StickyContext from './StickyContext'
+
+/**
+ * Sticky content stays fixed to the browser viewport while
+ * another column of content is visible on the page.
+ */
+export default class Sticky extends Component {
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Offset in pixels from the top of the screen when fixing element to viewport. */
+    offset: PropTypes.number,
+
+    /** Offset in pixels from the bottom of the screen when fixing element to viewport. */
+    bottomOffset: PropTypes.number,
+
+    /** Whether element should be "pushed" by the viewport, attaching to the bottom of the screen when scrolling up. */
+    pushing: PropTypes.bool,
+
+    /* Callback when element is bound to bottom of parent container. */
+    onBottom: PropTypes.func,
+
+    /**
+     * Callback when element is fixed to page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     * */
+    onStick: PropTypes.func,
+
+    /* Callback when element is bound to top of parent container. */
+    onTop: PropTypes.func,
+
+    /**
+     * Callback when element is unfixed from page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     * */
+    onUnstick: PropTypes.func,
+  }
+
+  static defaultProps = {
+    offset: 0,
+    bottomOffset: 0,
+  }
+
+  static _meta = {
+    name: 'Sticky',
+    type: META.TYPES.MODULE,
+  }
+
+  static contextTypes = {
+    contextEl: PropTypes.node,
+  }
+
+  static Context = StickyContext
+
+  state = {}
+
+  componentDidMount() {
+    this.contextEl = this.getContextEl()
+    this.update()
+    window.addEventListener('scroll', this.update)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.update)
+  }
+
+  getContextEl() {
+    let parent = this.triggerRef.parentElement
+
+    while (!(
+      (parent === document.body) ||
+      (parent.classList.contains('ui') && parent.classList.contains('context'))
+    )) {
+      parent = parent.parentElement
+
+      if (!parent) {
+        parent = document.body
+      }
+    }
+
+    return parent
+  }
```

Comment:
I think we can that use there [closest()](https://developer.mozilla.org/ru/docs/Web/API/Element/closest) with polyfill.